packages/expect-puppeteer/src/matchers/notToMatch.test.js

```diff
@@ -5,43 +5,51 @@ describe('not.toMatch', () => {
     await page.goto(`http://localhost:${process.env.TEST_SERVER_PORT}`)
   })
 
-  describe.each(['Page', 'Frame'])('%s', (pageType) => {
-    let page
-    setupPage(pageType, ({ currentPage }) => {
-      page = currentPage
-    })
-    it('should be ok if text is not in the page', async () => {
-      await expect(page).not.toMatch('Nop!')
-    })
-
-    it('should return an error if text is in the page', async () => {
-      expect.assertions(3)
-
-      try {
-        await expect(page).not.toMatch('home')
-      } catch (error) {
-        expect(error.message).toMatch('Text found "home"')
-        expect(error.message).toMatch('waiting for function failed')
-      }
-    })
-  })
+  describe.each(['Page', 'Frame', 'ShadowPage', 'ShadowFrame'])(
+    '%s',
+    (pageType) => {
+      let page
+      setupPage(pageType, ({ currentPage }) => {
+        page = currentPage
+      })
 
-  describe('ElementHandle', () => {
-    it('should be ok if text is in the page', async () => {
-      const dialogBtn = await page.$('#dialog-btn')
-      await expect(dialogBtn).not.toMatch('Nop')
-    })
-
-    it('should return an error if text is not in the page', async () => {
-      expect.assertions(3)
-      const dialogBtn = await page.$('#dialog-btn')
-
-      try {
-        await expect(dialogBtn).not.toMatch('Open dialog')
-      } catch (error) {
-        expect(error.message).toMatch('Text found "Open dialog"')
-        expect(error.message).toMatch('waiting for function failed')
-      }
-    })
-  })
+      const options = ['ShadowPage', 'ShadowFrame'].includes(pageType)
+        ? { traverseShadowRoots: true }
+        : {}
+
+      it('should be ok if text is not in the page', async () => {
+        await expect(page).not.toMatch('Nop!', options)
+      })
+
+      it('should return an error if text is in the page', async () => {
+        expect.assertions(3)
+
+        try {
+          await expect(page).not.toMatch('home', options)
+        } catch (error) {
+          expect(error.message).toMatch('Text found "home"')
+          expect(error.message).toMatch('waiting for function failed')
+        }
+      })
+
+      describe('ElementHandle', () => {
```

Comment:
moved `ElementHandle` into the page type block. Was there a reason it was not tested for frames?